src/NServiceBus.Metrics/RawData/OccurrenceWriter.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.IO;
+
+    static class OccurrenceWriter
+    {
+        const long Version = 1;
+
+        // WIRE FORMAT, Version: 1
+
+        //0                   1                   2                   3
+        //0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
```

Comment:
Ok. I followed your advice and tried to attack the current approach.

Being given an endpoint that processes 1000 msg a sec, 1 every ms. If we assume that dispatching a message takes 10ms, then the reporter would send data every 10ms, sending 100msg per second. We can bound it by introducing a delay when dispatching data, with a TimeSpan (let's say) 100ms. This would mean that, we will check the buffer every 100ms, if something is in there and send to the other side.

Would this boundary lower the number of possible messages? Yes.

Another option would be to check for the number of messages in the buffer, or bulk them up in the writer. 

What do you think @danielmarbach ?